R/conditions.R

```diff
@@ -130,6 +130,59 @@ captureStackTraces <- function(expr) {
 #' @include globals.R
 .globals$deepStack <- NULL
 
+getCallStackDigest <- function(callStack, warn = FALSE) {
+  dg <- attr(callStack, "digest", exact = TRUE)
+  if (!is.null(dg)) {
+    return(dg)
+  }
+
+  if (isTRUE(warn)) {
+    rlang::warn(
+      "Call stack doesn't have a cached digest; expensively computing one now",
+      .frequency = "once",
+      .frequency_id = "deepstack-uncached-digest-warning"
+    )
+  }
+
+  rlang::hash(getCallNames(callStack))
```

Comment:
Are we sure we don't want this to be `rlang::hash(callStack)`?  `getCallNames()` drops environments etc from the call, so potentially we'd get false positive matches?

Reply:
I guess it does make sense though in terms of not _showing_ the same stack multiple times?

Reply:
I started with hashing the call stack, it was _shockingly_ slow and also did not perform any deduplication. Using the names instead solved both problems. And yes, since this is all about managing the display of these stack traces, I do think if they print identically, we should treat them as identical.